presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaIntegrationSmokeTest.java

```diff
@@ -160,6 +162,119 @@ private KafkaTopicFieldDescription createOneFieldDescription(String name, Type t
+    public final Object[][] testJsonDateTimeFormatsDataProvider()
+    {
+        return testJsonDateTimeFormatsData().stream()
+                .collect(toDataProvider());
+    }
+
+    private List<JsonDateTimeTestCase> testJsonDateTimeFormatsData()
+    {
+        return ImmutableList.<JsonDateTimeTestCase>builder()
+                .add(new JsonDateTimeTestCase(
+                        "custom_date_time",
+                        ImmutableList.of("f_time", "f_time_with_time_zone", "f_timestamp", "f_timestamp_with_time_zone", "f_date"),
+                        ImmutableList.of(
+                                ImmutableList.of("TIME '01:02:03.456'",
+                                        "TIME '01:02:03.456 UTC'",
+                                        "TIMESTAMP '2020-07-15 01:02:03.456'",
+                                        "TIMESTAMP '2020-07-15 01:02:03.456 UTC'",
+                                        "DATE '2020-07-15'"))))
+                .add(new JsonDateTimeTestCase(
+                        "iso8601",
+                        ImmutableList.of("f_time", "f_time_with_time_zone", "f_timestamp", "f_timestamp_with_time_zone", "f_date"),
+                        ImmutableList.of(
+                                ImmutableList.of("TIME '01:02:03.456'",
+                                        "TIME '01:02:03.456 UTC'",
+                                        "TIMESTAMP '2020-07-15 01:02:03.456'",
+                                        "TIMESTAMP '2020-07-15 01:02:03.456 UTC'",
+                                        "DATE '2020-07-15'"))))
+                .add(new JsonDateTimeTestCase(
+                        "rfc2822",
+                        ImmutableList.of("f_time", "f_time_with_time_zone", "f_timestamp", "f_timestamp_with_time_zone", "f_date"),
+                        ImmutableList.of(
+                                ImmutableList.of("TIME '01:02:03'",
+                                        "TIME '01:02:03 UTC'",
+                                        "TIMESTAMP '2020-07-15 01:02:03'",
+                                        "TIMESTAMP '2020-07-15 01:02:03 UTC'",
+                                        "DATE '2020-07-15'"))))
+                .add(new JsonDateTimeTestCase(
+                        "milliseconds_since_epoch",
+                        ImmutableList.of("f_time", "f_time_with_time_zone", "f_timestamp", "f_timestamp_with_time_zone"),
+                        ImmutableList.of(
+                                ImmutableList.of("TIME '01:02:03.456'",
+                                        "TIME '01:02:03.456 UTC'",
+                                        "TIMESTAMP '2020-07-15 01:02:03.456'",
+                                        "TIMESTAMP '2020-07-15 01:02:03.456 UTC'"))))
+                .add(new JsonDateTimeTestCase(
+                        "seconds_since_epoch",
+                        ImmutableList.of("f_time", "f_time_with_time_zone", "f_timestamp", "f_timestamp_with_time_zone"),
```

Comment:
Since the column list is the same for each test case, you can hard-code it in the test.

Reply:
unfortunately seconds/milliseconds since epoch don't support date